js/source/geojson_source.js

```diff
@@ -41,28 +44,13 @@ module.exports = GeoJSONSource;
  * map.removeSource('some id');  // remove
  */
 function GeoJSONSource(options) {
-    options = options || {};
+    this.options = options = options || {};
```

Comment:
I prefer to not add additional uses of `this.options` in the sources, but don't demand that you fix existing ones :-)